### PR TITLE
feat(framework) Introduce `flwr-simulation` process

### PR DESCRIPTION
**Pull request by jafermarq**
- Introduces `flwr-simulation`
- Moves previous content of `simulation/app.py` into `simulation/legacy_app.py`

Merge after: https://github.com/adap/flower/pull/4448

### CODE REVIEW

File: src/py/flwr/simulation/app.py [hunk 1]
```diff
@@ -12,371 +12,299 @@
...
+            # Change status to Running
+            run_status_proto = run_status_to_proto(RunStatus(Status.RUNNING, "", ""))
+            conn._stub.UpdateRunStatus(
+                UpdateRunStatusRequest(run_id=run.run_id, run_status=run_status_proto)
+            )
+
+            # Pull Federation Options
+            fed_opt_res: GetFederationOptionsResponse = conn._stub.GetFederationOptions(
+                GetFederationOptionsRequest(run_id=run.run_id)
+            )
```
Comment (panh99) [inline src/py/flwr/simulation/app.py:247]: May I know the purpose of thie `GetFederationOptions` RPC? Can we merge it into `PullSimulationInputs`?

**Review by panh99 — APPROVED** [review]
Looks good to me! Excited to see how similar it is to `flwr-serverapp`.

File: src/py/flwr/simulation/app.py (hunk 1 above)
Comment (jafermarq) [reply src/py/flwr/simulation/app.py:247]: valid point. We can do the merging you propose after this and https://github.com/adap/flower/pull/4462 get merged I suppose.

File: src/py/flwr/simulation/app.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+                continue
+
+            context = context_from_proto(res.context)
+            run = run_from_proto(res.run)
+            fab = fab_from_proto(res.fab)
+
+            # Start log uploader for this run
+            log_uploader = start_log_uploader(
+                log_queue=log_queue,
+                node_id=0,
```
Comment (danieljanes) [inline src/py/flwr/simulation/app.py:203]:
```suggestion
                node_id=context.node_id,
```

**Pull request merged**